ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackendFactory.scala

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend
+
+import com.daml.platform.store.DbType
+import com.daml.platform.store.backend.h2.H2StorageBackendFactory
+import com.daml.platform.store.backend.oracle.OracleStorageBackendFactory
+import com.daml.platform.store.backend.postgresql.PostgresStorageBackendFactory
+
+trait StorageBackendFactory {
+  def ingestionStorageBackend: IngestionStorageBackend[_]
+  def parameterStorageBackend: ParameterStorageBackend
+  def configurationStorageBackend: ConfigurationStorageBackend
+  def partyStorageBackend: PartyStorageBackend
+  def packageStorageBackend: PackageStorageBackend
+  def deduplicationStorageBackend: DeduplicationStorageBackend
+  def completionStorageBackend: CompletionStorageBackend
+  def contractStorageBackend: ContractStorageBackend
+  def eventStorageBackend: EventStorageBackend
+  def dataSourceStorageBackend: DataSourceStorageBackend
+  def DBLockStorageBackend: DBLockStorageBackend
+  def integrityStorageBackend: IntegrityStorageBackend
+  def resetStorageBackend: ResetStorageBackend
+}
+
+object StorageBackendFactory {
+  def of(dbType: DbType): StorageBackendFactory =
+    dbType match {
+      case DbType.H2Database => H2StorageBackendFactory
+      case DbType.Postgres => PostgresStorageBackendFactory
+      case DbType.Oracle => OracleStorageBackendFactory
+    }
+
+  def readStorageBackendFor(dbType: DbType): ReadStorageBackend = {
+    val factory = of(dbType)
+    ReadStorageBackend(
+      configurationStorageBackend = factory.configurationStorageBackend,
+      partyStorageBackend = factory.partyStorageBackend,
+      packageStorageBackend = factory.packageStorageBackend,
+      completionStorageBackend = factory.completionStorageBackend,
+      contractStorageBackend = factory.contractStorageBackend,
+      eventStorageBackend = factory.eventStorageBackend,
+    )
+  }
+}
+
+case class ReadStorageBackend(
```

Comment:
this grouping motivated by client code (JdbcLedgerDao) for convenience. Let's see how it will look finally in the upcoming changes: we might decide to remove it.